### PR TITLE
Migrate to SnakeYAML Engine

### PR DESCRIPTION
**Pull request by asomov**
Migrate to SnakeYAML Engine:

- to support YAML 1.2
- to get rid of false positives for potential vulnerabilities in SnakeYAML

### CODE REVIEW

**Comment by asomov** [comment]
I applied this, because I could not build it otherwise (it was `false`):
`<japicmp.skip>true</japicmp.skip>`

**Review by kinow — COMMENTED** [review]
Hi @asomov , thanks for the PR.

I've enabled the workflow to see what's the status of the build.

I haven't followed much of the development in these Java YAML libraries, but this task would require a JIRA issue too so that others can comment here/there, and also for the changelog. Could you create one, please?

And since it changes the public API, I believe we would have to assign it to a major release.

If you have more updates to this PR, please push your changes to the same branch so this PR is updated, instead of closing and creating new PR's, since that means someone will have to allow the workflow to run (as it's your first contribution to this repository - GitHub Actions security) and also since the previous comments and reviews would be discarded when this PR is closed.

Thank you!
Bruno

File: pom.xml [hunk 1]
```diff
@@ -551,15 +551,15 @@
...
+      org.snakeyaml.engine.*;resolution:=optional,
       *
     </commons.osgi.import>
     <log4j.version>2.19.0</log4j.version>
     <slf4j.version>2.0.5</slf4j.version>
     <!-- Spring 6 requires Java 17 -->
     <spring.version>5.3.24</spring.version>
 
-    <japicmp.skip>false</japicmp.skip>
+    <japicmp.skip>true</japicmp.skip>
```
Comment (garydgregory) [inline pom.xml:562]: -1 obviously, you can't turn off this check for binary compatibility.

**Review by garydgregory — CHANGES_REQUESTED** [review]
See comment.

File: pom.xml (hunk 1 above)
Comment (asomov) [reply pom.xml:562]: it is binary incompatible

**Comment by asomov** [comment]
@kinow I did my best to create a JIRA, but unfortunately there is no way to create an account there. I am stuck

File: pom.xml (hunk 1 above)
Comment (garydgregory) [reply pom.xml:562]: Yes: Don't change that setting. You cannot break binary compatibility (BC) outside of a major release upgrade. If PR breaks BC, then it's -1.

Gary

Comment (asomov) [reply pom.xml:562]: @garydgregory how can I indicate that it is a backwards incompatible change ? To be included in the next major release. I think the master branch is exactly for it

**Comment by garydgregory** [comment]
You don't indicate anything, and you don't turn off JApiCmd as JApiCmd will fail the build if the PR breaks BC.

File: pom.xml (hunk 1 above)
Comment (garydgregory) [reply pom.xml:562]: You don't indicate anything, and you don't turn off JApiCmd as JApiCmd will fail the build if the PR breaks BC.

Comment (asomov) [reply pom.xml:562]: @garydgregory I did not quite catch you. Do you mean that I need to provide a PR which breaks the build ?

**Comment by garydgregory** [comment]
Since this PR breaks BC according to your earlier comment, it is -1. This PR hides this fact by disabling BC checking. BC can only be broken in a major release. There are currently no plans for a major release. My advice is to change this PR by reenabling BC checking and leave it open as a reminder for the next major release. Creating a Jira would also serve as a reminder for this request.

**Comment by asomov** [comment]
@garydgregory I think the simplest for me is keep this PR until a major version comes.

**Review by garydgregory — CHANGES_REQUESTED** [review]
I am marking this as -1 and "request change" in GH to fix BC checking. We cannot keep this PR open as a green build PR in case it gets merged by accident. It can stay open with a failing build.

**Comment by garydgregory** [comment]
We updated snakeyaml to 2.0.

**Comment by garydgregory** [comment]
@asomov 
Closing: We updated snakeyaml to 2.0.

**Pull request closed without merging**